subgraphs/_reference_/src/sdk/protocols/perpfutures/poolSnapshot.ts

```diff
@@ -0,0 +1,481 @@
+    snapshot.dailyActiveBorrowers = previousSnapshot
+      ? snapshot.cumulativeUniqueBorrowers -
+        previousSnapshot.cumulativeUniqueBorrowers
+      : snapshot.cumulativeUniqueBorrowers;
+
+    snapshot.cumulativeUniqueLiquidators =
+      this.pool.cumulativeUniqueLiquidators;
+    snapshot.dailyActiveLiquidators = previousSnapshot
+      ? snapshot.cumulativeUniqueLiquidators -
+        previousSnapshot.cumulativeUniqueLiquidators
+      : snapshot.cumulativeUniqueLiquidators;
+
+    snapshot.cumulativeUniqueLiquidatees =
+      this.pool.cumulativeUniqueLiquidatees;
+    snapshot.dailyActiveLiquidatees = previousSnapshot
+      ? snapshot.cumulativeUniqueLiquidatees -
+        previousSnapshot.cumulativeUniqueLiquidatees
+      : snapshot.cumulativeUniqueLiquidatees;
+
+    snapshot.longPositionCount = this.pool.longPositionCount;
+    snapshot.dailyLongPositionCount = previousSnapshot
+      ? snapshot.longPositionCount - previousSnapshot.longPositionCount
+      : snapshot.longPositionCount;
+
+    snapshot.shortPositionCount = this.pool.shortPositionCount;
+    snapshot.dailyShortPositionCount = previousSnapshot
+      ? snapshot.shortPositionCount - previousSnapshot.shortPositionCount
+      : snapshot.shortPositionCount;
+
+    snapshot.openPositionCount = this.pool.openPositionCount;
+    snapshot.dailyOpenPositionCount = previousSnapshot
+      ? snapshot.openPositionCount - previousSnapshot.openPositionCount
+      : snapshot.openPositionCount;
+
+    snapshot.closedPositionCount = this.pool.closedPositionCount;
+    snapshot.dailyClosedPositionCount = previousSnapshot
+      ? snapshot.closedPositionCount - previousSnapshot.closedPositionCount
+      : snapshot.closedPositionCount;
+
+    snapshot.cumulativePositionCount = this.pool.cumulativePositionCount;
+    snapshot.dailyCumulativePositionCount = previousSnapshot
+      ? snapshot.cumulativePositionCount -
+        previousSnapshot.cumulativePositionCount
+      : snapshot.cumulativePositionCount;
+
+    snapshot.inputTokenBalances = this.pool.inputTokenBalances;
+    snapshot.inputTokenWeights = this.pool.inputTokenWeights;
+    snapshot.outputTokenSupply = this.pool.outputTokenSupply;
+    snapshot.outputTokenPriceUSD = this.pool.outputTokenPriceUSD;
+    snapshot.stakedOutputTokenAmount = this.pool.stakedOutputTokenAmount;
+    snapshot.rewardTokenEmissionsAmount = this.pool.rewardTokenEmissionsAmount;
+    snapshot.rewardTokenEmissionsUSD = this.pool.rewardTokenEmissionsUSD;
+
+    snapshot.save();
+  }
+
+  getVolumeHourlyTracker(): _VolumeHourlyTracker {
+    const hourID =
+      this.pool._lastUpdateTimestamp!.toI32() / constants.SECONDS_PER_HOUR;
+
+    return getOrCreateVolumeHourlyTracker(hourID, this.pool);
+  }
+
+  getVolumeDailyTracker(): _VolumeDailyTracker {
+    const dayID =
+      this.pool._lastUpdateTimestamp!.toI32() / constants.SECONDS_PER_DAY;
+
+    return getOrCreateVolumeDailyTracker(dayID, this.pool);
+  }
+}
```

Comment:
Why `dailyVolumeUSD` is calculated comparing cumulatives, but `dailyVolumeByToken` is done via a tracker? Is there a reason that prevents us from doing it in a unique way for both?

Reply:
`dailyVolumeByToken` is the total volume of a specific token in a pool on a daily basis. We cannot use the cumulative here because it is calculated using the sum of the volume of all tokens in a pool, that's why I have created a separate tracker for this.

Reply:
You could keep an array of cumulatives :D